dip-0008.md

```diff
@@ -275,6 +277,109 @@ resolve any ambiguities which might occur in the last few blocks.
 If the need arises to include block signatures in initial block download, we
 will update this DIP and implementations accordingly.
 
+## Calculations
+
+We consider the scenario where an attacker has assumed control of a number of masternodes. We will then calculate two probabilities that the attacker will
+be able to disrupt the network. Specifically, we will calculate the probability
+that an attacker will be able to prevent all ChainLocks for a twelve hour period, and
+we will calculate the probability that an attacker can successfully create a malicious
+or erroneous ChainLock.
+
+Quorum selection is designed to provide for an effectively simple random
+choice of masternodes in each quorum. Thus we may assume that each quorum is
+a simple random selection. If there are a total of 'N' masternodes the
+number of possible quorums is the number of ways that 400 objects can be
+chosen out of `N`. In some literature this number is
+written <sub>`N`</sub>`C`<sub>`400`</sub> and is enunciated as
+'N choose 400'. Numbers that arise in this fashion are all
+binomial coefficients and have a delightful relation to Pascal's Triangle.
+
+For any process with an observable outcome the set of all outcomes is
+called a sample space.  If all outcomes are equally likely, as in the
+selection of a quorum, the probability of any event is the number of
+outcomes that satisfy that event over the total number outcomes.
+
+An attacker can temporarily prevent ChainLocks if they control
+160 out of 400 masternodes in the quorum.  If the attacker controls
```

Comment:
Wouldn't they actually need to control 161? 240/400 could still construct a signature.

Reply:
Well ... yes.  Good catch.

Reply:
commit 1f622c903aefff6cfc8df6dcab202e74bf59f05a should address this.  

In a previous commit I restructured the table and the calculations were changed there.